### PR TITLE
Clan System Part 2 - UI

### LINKED ISSUE #3639: clan system
Have clan ownership / ability to transfer leadership
Allow clans to be closed/invite only
User can be member of multiple clans
Person can be leader of multiple clans

### PR DESCRIPTION
**Pull request by ryanbarlow97**
## Description:

Continuation from #3276 

Adds the complete client-side clan UI as a Lit web component (`<clan-modal>`), a typed API client with Zod-validated responses, shared response schemas, and a reusable `<confirm-dialog>` component. 

### New: `ClanModal.ts`

| View | What it does |
|------|-------------|
| **My Clans** | Lists joined clans + pending join requests (built from `/users/@me`, no extra fetches) |
| **Browse** | Search by tag (min 3 chars), paginated results, configurable per-page (10/25/50) |
| **Clan Detail** | Stats, paginated + searchable member list, role badges, join/leave/request actions |
| **Manage** | Edit name (max 35 chars) + description, toggle open/invite-only, disband |
| **Transfer** | Leadership transfer with member selector + confirmation |
| **Requests** | Approve/deny join requests (leader/officer) |
| **Bans** | View and unban (leader/officer) |
| **My Requests** | View and withdraw outgoing requests |

### New: `ConfirmDialog.ts`

Reusable `<confirm-dialog>` Lit component — replaces native `confirm()`/`prompt()` which are blocked or broken on mobile and CrazyGames iframes. Supports danger/warning variants and an optional textarea (used for ban reasons). Fires `confirm`/`cancel` events.

### New: `ClanApi.ts`

Typed API client covering all clan endpoints. Every response is Zod-validated. Auth header is always last in the spread (can't be overridden by callers). Unknown server error messages always fall back to a generic client-side string — never displayed verbatim.

### New: `ClanApiSchemas.ts` (in `src/core/`)

Shared Zod schemas for clan API responses with max-length constraints on `name` (35) and `description` (200). Lives in `core/` so it can be consumed by both client code and the leaderboard table.

### Modified: `ApiSchemas.ts`

- Added `clans` and `clanRequests` arrays to `UserMeResponseSchema`
- Moved clan leaderboard schemas out to `ClanApiSchemas.ts`
- Renamed `LeaderboardClanTagSchema` → `RequiredClanTagSchema`

### Modified: `Api.ts`

- Added `invalidateUserMe()` to bust the cached `/users/me` response after mutations
- Removed `fetchClanLeaderboard` (moved to `ClanApi.ts`)

### Tests

- `ClanModal.test.ts` — rendering, view navigation, user actions
- `ClanApiQueries.test.ts` — fetch functions, error handling, pagination
- `ClanApiMutations.test.ts` — join, leave, kick, ban, promote, transfer, etc.
- `ClanApiBans.test.ts` — ban/unban calls and error paths
- `ClanApiSchemas.test.ts` — Zod schema validation edge cases
- `LeaderboardModal.test.ts` — updated imports

## Notable design decisions

- **Not-logged-in state** — shows "Sign in to join clans" instead of false "no clans" empty state
- **Rate limit feedback** — reads `Retry-After` header and surfaces wait time to the user

## Please complete the following:

- [x] I have added screenshots for all UI updates
- [x] I process any text displayed to the user through translateText() and I've added it to the en.json file
- [x] I have added relevant tests to the test directory
- [x] I confirm I have thoroughly tested these changes and take full responsibility for any bugs introduced

## Please put your Discord username so you can be contacted if a bug or regression is found:

w.o.n

### CODE REVIEW

File: src/client/Api.ts [hunk 1]
```diff
@@ -288,3 +310,400 @@ export async function getNews(): Promise<NewsItem[]> {
...
+    const json = await res.json();
+    const parsed = ClanMembersResponseSchema.safeParse(json);
+    if (!parsed.success) {
+      console.warn("fetchClanMembers: Zod validation failed", parsed.error);
+      return false;
+    }
+    return parsed.data;
+  } catch {
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (ryanbarlow97) [reply src/client/Api.ts]: in practice myClanRoles (from the @me endpoint) is the primary source of role truth

[the remark this answers is not in the document]
Comment (ryanbarlow97) [reply src/client/Api.ts]: @coderabbitai

File: src/client/ClanModal.ts [hunk 2]
```diff
@@ -0,0 +1,1998 @@
...
+      this.selectedClan.tag,
+      page,
+      this.membersPerPage,
+    );
+    if (res) {
+      this.members = res.results;
+      this.membersTotal = res.total;
+      this.memberPage = page;
+      this.pendingRequestCount = res.pendingRequests ?? 0;
+    }
```
[the remark this answers is not in the document]
Comment (ryanbarlow97) [reply src/client/ClanModal.ts]: Already addressed

File: src/client/ClanApi.ts [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+    if (res.status === 409) return { error: "clan_modal.error_already_member" };
+    if (res.status === 429) return { error: "clan_modal.error_rate_limited" };
+    if (!res.ok) {
+      const body = await res.json().catch(() => ({}));
+      return {
+        error:
+          (body as { message?: string }).message ?? "clan_modal.error_failed",
+      };
+    }
+    return (await res.json()) as { status: "joined" | "requested" };
```
[the remark this answers is not in the document]
Comment (ryanbarlow97) [reply src/client/ClanApi.ts]: check again

[the remark this answers is not in the document]
Comment (ryanbarlow97) [reply src/client/ClanApi.ts]: server endpoint is ours and returns exactly { status: "joined" | "requested" }.

**Comment by nicolas-machugounot** [comment]
Hey how can i help ?

**Comment by ryanbarlow97** [comment]
> Hey how can i help ?

I'm all good thanks, this is a pr not an issue

**Comment by evanpelle** [comment]
can you update the description?

**Review by Duwibi — DISMISSED** [review]

File: src/client/ClanModal.ts
Comment (evanpelle) [inline src/client/ClanModal.ts:1]: this file is way too large. please split it out into lit components, put them in src/client/components/clan/...

let's try to keep it under 500 lines of code.

Comment (ryanbarlow97) [reply src/client/ClanModal.ts:1]: should be a lot better now

**Comment by MitchZinck** [comment]
Closes #2978

**Review by evanpelle — APPROVED** [review]
thanks!
